### DISCUSSION TITLE
Adding/changing programs in rom

### DISCUSSION THREAD

**Discussion by IvoLeal72** [cc-tweaked/CC-Tweaked#1201] [Q&A] [answered]
Hello, is there a documentation on the various ways to do this (on a server I own)?
I found people saying a folder in the server works, others saying a resource pack.

But I'd like some guide or wiki

**Comment by Wojbie** [comment, marked as answer]
On current CC:T versions CC uses data packs for adding stuff to rom. You can find example data-pack here: https://github.com/cc-tweaked/datapack-example

**Reply by IvoLeal72** [reply]
What if I want to create an API available in every computer, that requires java, is this possible or do I always need a new peripheral?

**Reply by Wojbie** [reply]
If you put os.loadAPI style api in `lua/rom/apis/` folder it will be auto-loaded into _G on startup of computer.
If you put require style module into `lua/rom/modules/main` or `lua/rom/modules//turtle` folder it will be available on all computers or only on turtles.
If you put a program into `lua/rom/autorun/` it will be auto-executed before computer startup code is ran.
If you just want program to be available to end user then that's what `lua/rom/programs/` and `lua/rom/programs/turtl`e folders are for.
If you look at README of data-pack example i linked before it explains it in more details.

**Reply by IvoLeal72** [reply]
That works for 100% lua apis, I got that, my question is if I want to make a mod that adds an java/lua api to cc, but without a peripheral, is it possible, or do I always need a peripheral?

**Reply by SquidDev** [reply]
You can do that with [`ILuaAPI`](https://tweaked.cc/javadoc/)
